src/praisonai-agents/praisonaiagents/skills/validator.py

```
@@ -191,9 +191,11 @@ def validate(skill_dir: Path, strict: bool = False) -> List[str]:
         return ["Missing required file: SKILL.md"]
 
     try:
-        content = skill_md.read_text()
+        content = skill_md.read_text(encoding="utf-8")
         metadata, _ = parse_frontmatter(content)
-    except ParseError as e:
+    except (ParseError, UnicodeDecodeError) as e:
         return [str(e)]
+    except UnicodeDecodeError as e:
```

Ruff (0.15.13)

[warning] try-except block with duplicate exception `UnicodeDecodeError`

(B025)